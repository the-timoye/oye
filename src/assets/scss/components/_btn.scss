.btn {
    display: inline-block;
    padding: .4rem 2rem;
    border: 1px solid $beige;
    border-radius: 20px;
    text-align: center;
    margin: 0 .5rem;
    transition: all .3s ease-in-out;

    &-primary {

        color: $light_green;
        background: $green;
        &:hover {        
            background: $beige;
            color: $green;
        }
    }
    &-secondary {
        background: $light_green;
        color: $green;
        &:hover {        
            background: $beige;
            
        }
    }
}