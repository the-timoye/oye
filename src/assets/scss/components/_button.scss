.button {
    padding: 0.5rem 2.5rem;
    font-weight: 400;
    border-radius: 5px;

    &-secondary {
        border: 1px solid $primary;
        background-color: transparent;
        color: $primary;
    }
}
