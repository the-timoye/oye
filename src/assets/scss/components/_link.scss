a.link {
    display: inline-block;
    margin: 1rem auto;
    padding: 0.5rem 0;

    hr.link__hr {
        margin-top: 0.5rem;
        border: 0;
        height: 1px;

        &.__light {
            background-image: linear-gradient(to right, rgba(0, 0, 0, 0), $text, rgba(0, 0, 0, 0));
        }
        &.__dark {
            background-image: linear-gradient(to right, rgba(0, 0, 0, 0), $primary, rgba(0, 0, 0, 0));
        }
    }

    &.__light {
        color: $text;
    }
    &.__dark {
        color: $primary;
    }
}
