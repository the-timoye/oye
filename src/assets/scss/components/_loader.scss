.lds-ripple {
    display: inline-block;
    position: absolute;
    top: 50%;
    left: 50%;
    width: 80px;
    height: 80px;

    div {
        position: absolute;
        border: 4px solid $text;
        opacity: 1;
        border-radius: 50%;
        animation: lds-ripple 1s cubic-bezier(0, 0.2, 0.8, 1) infinite;

        &:nth-child(2) {
            animation-delay: -0.5s;
        }
    }
}
