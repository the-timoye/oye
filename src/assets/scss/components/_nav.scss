nav {
    display: flex;
    justify-content: space-between;
    padding: 2rem 1.5rem;

    img {
        height: 3rem;
    }

    div {
        & :not(:last-child) {
            margin-right: 1rem;
        }
        a {
            cursor: pointer;
        }
    }
}
