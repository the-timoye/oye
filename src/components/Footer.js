import React from 'react';
import chatIcon from '../assets/images/chat.png';

export const Footer = () => {
    return (

        <footer className="footer">
            <span>Copyright &#169; 2020</span>
            <span>Designed and Developed by <a href="/"> thetimoye </a> &#9825;</span>
            <img src={chatIcon} alt='chat' />
        </footer>

    )
}