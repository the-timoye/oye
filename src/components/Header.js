import React, { useState, useEffect } from "react";
import headerImg from "../assets/images/header.png";
import IG from "../assets/images/instagram.png";
import gitHub from "../assets/images/github.png";
import relpIt from "../assets/images/replit.png";
import { typingAnimation } from "../utility/helpers";

export const Header = () => {
    const [state, setState] = useState({
        introTexts: ["Hi, I'm Timi, aka thetimoye", "I'm a fullstack dev"],
        intro: '',
        delay: 2000,
        bannerSrc: headerImg,
        mainIntro: "Hi, I'm Timi",
        subIntro: 'Iâ€™m a Software Developer with deep love for code, and everything there is to know about Software Development and Data Engineering',
        socialHandles: [
            { image: gitHub, url: 'https://github.com/the-timoye', name: 'Git Hub' },
            { image: IG, url: 'https://instagram.com/thetimoye', name: 'Instagram' },
            { image: relpIt, url: 'https://repl.it/@thetimoye', name: 'Repl It' }
        ]
    });


    const { bannerSrc, intro, subIntro, socialHandles } = state;

    useEffect(() => {
        if (state.introTexts.length > 0) {
            typingAnimation(setState, state.introTexts, state.delay);

        }
    }, [state.introTexts, state.delay]);

    return (
        <header className="home">
            <div className="intro">
                <h2 className={state.intro === '' ? 'mb-5' : ''} > {state.intro} </h2>
                <p> {subIntro} </p>
                <>
                    {socialHandles.map(handle => {
                        return (
                            <a href={handle.url} target="_blank" rel="noreferrer" key={handle.name}> <img className='socials' src={handle.image} alt={handle.name} /> </a>
                        )
                    })}
                </>
            </div>
            <img src={bannerSrc} alt="home-header" className='banner' />
        </header>
    )
}