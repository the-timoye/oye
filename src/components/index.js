export * from './Footer';
export * from "./Header";
export * from "./Hobbies";
export * from "./Nav";
export * from "./Projects";
export * from "./SectionTitle"
export * from "./Skills";