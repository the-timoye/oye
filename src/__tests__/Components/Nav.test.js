import React from 'react';
import renderer from 'react-test-renderer';
import { Nav } from '../../components';


describe('Navigation Bar Tests', () => {
    const NavRenderer = renderer.create(<Nav />);
    const NavInstance = NavRenderer.root;
    // NavInstance.instance

    it('renders correctly', () => {
        expect(NavRenderer).toMatchSnapshot();
    });

    it('should have a nav element with two children', () => {
        const nav = NavInstance.findByType('nav');
        expect(nav).toBeTruthy();
        expect(nav.children.length).toBeGreaterThan(1);
    });

    it('should contain a logo', () => {
        const logo = NavInstance.findByProps({ className: 'logo' });
        expect(logo).toBeTruthy();
    });

    it('should have alt texts for images', () => {
        const navLinksContainer = NavInstance.findByType('div');
        const images = navLinksContainer.findAllByType('img');

        expect(images.length).toBeGreaterThan(1);
        images.forEach(image => {
            expect(image.props.alt).toBeTruthy();
        });
    });

    it('tests nav links', () => {
        const navLinksContainer = NavInstance.findByType('div');
        const links = navLinksContainer.findAllByType('a');
        const cvLink = links[0];

        links.forEach(image => {
            expect(image.props.href).toBeTruthy();
        });
        expect(navLinksContainer.children.length).toBeGreaterThan(1);
        expect(cvLink.props.rel).toEqual('noreferrer');
    })

});