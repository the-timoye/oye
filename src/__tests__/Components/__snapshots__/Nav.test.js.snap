// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Bar Tests renders correctly 1`] = `
<nav>
  <img
    alt="logo"
    className="logo"
    src="logo.png"
  />
  <div>
    <a
      href="Oluyinka Oluwatimilehin.pdf"
      onClick={[Function]}
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="file"
        src="file.png"
      />
    </a>
    <a
      href="mailto:thetimoye@gmail.com"
    >
      <img
        alt="mail"
        src="mail.png"
      />
    </a>
  </div>
</nav>
`;
